sdks/java/extensions/sql/src/test/java/org/apache/beam/sdk/extensions/sql/meta/provider/pubsub/PubsubTableProviderIT.java

```diff
@@ -439,8 +440,20 @@ public void testSQLLimit() throws Exception {
                   return result.build();
                 });
 
-    eventsTopic.assertSubscriptionEventuallyCreated(
-        pipeline.getOptions().as(GcpOptions.class).getProject(), Duration.standardMinutes(5));
+    try {
+      eventsTopic.assertSubscriptionEventuallyCreated(
+          pipeline.getOptions().as(GcpOptions.class).getProject(), Duration.standardMinutes(5));
+    } catch (AssertionError assertionError) {
+      // If we timed out, check if the forked thread had an exception. If something
+      // did go wrong there this call will raise an ExecutionException, which we allow to bubble
+      // up.
+      try {
+        queryResult.get(1, TimeUnit.SECONDS);
```

Comment:
Could you just use `queryResult.get()` here? (Wouldn't the forked thread have an exception before the timeout?)

Reply:
The timeout is for the case where no exception was raised on the forked thread, but we still didn't create a subscription for some reason. If we just call `get()` then we would hang indefinitely in that case. I changed the timeout to 0 seconds (which is maybe what you thought `get()` would do?), to make the intention more clear.